frontend/src/components/personal/EmailChangeForm.tsx

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable react/jsx-no-bind */
+/* eslint-disable react/jsx-sort-props */
+import React from 'react'
+import { useFormik } from 'formik'
+import { CardActions, Button, TextField} from '@material-ui/core'
+import * as Yup from 'yup'
+import {useSelector} from 'react-redux'
+
+import {useStyles, useStyleModal, getModalStyle} from './ProfilePreferencesPage.styles'
+import {RootState} from '../../reducers/index'
+import Modal from '@material-ui/core/Modal'
+
+
+export default function EmailChangeForm (): React.ReactElement {
+    const user = useSelector((state: RootState) => state.operatePreferencesDataReducer)
+
+    const classes = useStyles()
+    const modalClass = useStyleModal()
+    
+    const validationSchema = Yup.object({
```

Comment:
You can move validationSchema out of component (and file) so you will have an option to test it separately from component (and do not recreate for each render).

Reply:
moved to utils/ValidationSchemes for password and email schemes. By now not quite sure how to add comparing for field that exist only in email change form file (old and new email)